### DISCUSSION TITLE
Slug for root of pages directory

### DISCUSSION THREAD

**Discussion by jeorluk** [vercel/next.js#14686] [Help]
I'm pulling in dynamic pages from a CMS. In my pages directory I have [slug].js and I query all pages and generate them dynamically.  My question is what I would use for the slug if I just want the root (like /, index, etc.).  I'm just running locally right now, so what I want is to be able to put in localhost:3000 and get a dynamic page from the the CMS.

**Comment by retzion** [comment]
Have the same requirements. Any update?
I want to have something like
```
/pages/index.js (homepage hosted at / )
/pages/[slug].js (dynamic pages hosted at /<slug> )
```

**Reply by retzion** [reply]
Never mind. That schema works. My issue was the fallback function in my component.
```
  if (!router.isFallback) {
    return <ErrorPage statusCode={404} />
  }
```
